pkg/stub/handler.go

```diff
@@ -25,18 +27,28 @@ type Handler struct {
 }
 
 func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
-	if event.Deleted {
-		return nil
-	}
 	switch o := event.Object.(type) {
 	case *ingressv1alpha1.ClusterIngress:
-		return h.syncIngressUpdate(o)
+		if event.Deleted {
+			logrus.Infof("Deleting ClusterIngress object: %s", o.Name)
+			return nil
+		} else {
+			return h.syncIngressUpdate(o)
+		}
 	}
 	return nil
 }
 
 func (h *Handler) syncIngressUpdate(ci *ingressv1alpha1.ClusterIngress) error {
+	// Get the Owner Reference object for the current ClusterIngress
+	ciOwnerRef := getCIOwnerRef(ci)
+
 	ns, err := h.manifestFactory.RouterNamespace()
+	/* Set the Cluster Ingress Operator Object as Owner of the namespace
+	 * this allows the cascading deletion of all the objects when the
+	 * Cluster Ingress Operator object is deleted
+	 */
+	setOwnerRef(ns, ciOwnerRef)
```

Comment:
When more than one router is deployed (e.g. default and internal), deleting one router should not remove the entire router namespace. Multiple routers are sharing the same router namespace, serviceaccount, cluster roles and role bindings. 
I think you need to remove router namespace if there is only one router otherwise only remove daemonset and service which is specific to the deleted clusteringress custom resource. Is owner ref solves this use case?

Reply:
Yeah... and then, why bother deleting the namespace anyway?

Reply:
Each router would need its own namespace

Reply:
GC doesn't allow inter-namespace relationships, so owner refs are of no use when it comes to ClusterIngress-><resource set in router namespace>

Reply:
I think this is enough for now since we only have one router per ClusterIngress.

The use of ownerReferences pattern is the recommended way to delete objects in kubernetes: https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/

When we have other use cases we can easily refactor to deal with them using this pattern.

Reply:
We MUST support multiple routers in that namespace, it's a core feature and we already support it; if we stop supporting it that's a regression at this point. Regardless of the user guide, I know from working on GC with upstream that the use case may work sometimes but isn't supported. See recent commentary on this use case (namespaced owner, cluster scoped dependent): https://github.com/kubernetes/kubernetes/issues/65200

Relying on unsupported GC behavior caused one of our teams trouble last year when the gap was plugged, let's not set ourselves up for trouble.